jerry-main/main-unix.c

```diff
@@ -121,6 +121,33 @@ print_help (char *name)
                       name);
 } /* print_help */
 
+static void
+print_unhandled_exception (jerry_value_t error_value)
+{
+  assert (jerry_value_has_error_flag (error_value));
+
+  jerry_value_clear_error_flag (&error_value);
+  jerry_value_t err_str_val = jerry_value_to_string (error_value);
+  jerry_size_t err_str_size = jerry_get_string_size (err_str_val);
+  jerry_char_t err_str_buf[256];
+
+  if (err_str_size >= 256)
+  {
+    const char msg[] = "[Long message]";
+    err_str_size = sizeof (msg) / sizeof (char);
```

Comment:
Well, _this_ needs some adjustment, if I'm not mistaken. Correctly, this should be `err_str_size = sizeof (msg) / sizeof (char) - 1;`, as the current code does count the terminating zero as well. We don't want to copy that terminating zero and we want to get `err_str_buf[err_str_size] = 0;` at line 145 put that 0 to the correct index. (Right now, it would write a 0 after the already existing terminating zero.)

Example:
```c
#include <stdio.h>

int main()
{
    const char msg[] = "1234567";
    int size = sizeof(msg) / sizeof(char);
    printf("size(%s) = %d\n", msg, size);
}
```
prints: "size(1234567) = 8"